File: AM_12khz_wave_file.m
% script makes an am signal at 12khz carrier 
% output is a wave file 
% made by Thegmr140 on youtube  


close all
clear all


% read in audio signal 
[asig, aFs]  = audioread('c:\temp\audiosignal.wav');



%audio low pass filter, fc = 5khz 
fs = 48e3;
fc = 5e3;
fc = fc / (fs/2);
N = 1023;
hlpf = fir1(N,fc);


% get audio signal to 48khz sample rate 
if aFs ~= fs
    a = gcd(fs,aFs);
    p = fs / a;
    q = aFs / a;
    asig = resample(asig,p,q);
end


[nS,nch] = size(asig);


if nch == 2
    asig = asig(:,1) + asig(:,2);
    asig = asig.';
    asig = asig / max(asig);
else
    asig = asig(:,1);
    asig = asig.';
    asig = asig / max(asig);
end



asig = filter(hlpf,1,asig);



N = length(asig);
t = [1:N]/fs;
amsig = [1 + 1.2*asig] .* cos(2*pi*12e3*t);
amsig = amsig / max(amsig);




amsigwave = [    (amsig).'   ];         %  
audiowrite('c:\temp\AMSignal_12khz.wav' , amsigwave,  fs);















    
    
    
    
    
    
    

File: AM_12khz_wave_file.py
import numpy as np
from scipy.io import wavfile
from scipy.signal import fir_filter_design as ffd
from scipy.signal import lfilter, resample
import fractions

# read in audio signal, audio file needs to have 48khz sample rate!!!!
# wave file needs to be mono, not 2 ch stereo  
aFs , asig = wavfile.read('c:/temp/audiosignal.wav')

asig = asig.astype(np.float64)
asig = asig / np.max(asig)



# audio low pass filter, fc = 5khz
fs = 48000
fc = 5000
fc = fc / (fs / 2)
N = 1023
hlpf = ffd.firwin(N, fc)



asig = asig.T
asig = asig / np.max(asig)

asig = lfilter(hlpf, 1, asig)
N = len(asig)
t = np.arange(1, N + 1) / fs
amsig = (1 + 1.2 * asig) * np.cos(2 * np.pi * 12000 * t)
amsig = amsig / np.max(amsig)
amsigwave = amsig.T

wavfile.write('c:/temp/AMSignal_12khz.wav', fs, amsigwave.astype(np.float32))

